arduino/smartcar/smartcar.ino

```diff
@@ -14,18 +22,39 @@ void setup() {
 }
 
 void loop() {
-  car.setSpeed(90);
+  car.setSpeed(70); 
   preventCrash();
 }
 
-void preventCrash() {
-  unsigned int fDistance = front.getDistance();
-  if (fDistance > 0 && fDistance < 200) {
+void preventCrash(){
+  preventCrashWithUltrasonic(); 
+  preventCrashWithInfrared(frontIR, 0, 40); // Helps in short distances object detection.
+}
+
+void preventCrashWithUltrasonic() {
+  unsigned int distanceUS = frontUS.getDistance(); // min distance: 0.2 m, MAX distance: 4 m.
+  if (distanceUS > 20 && distanceUS < 50) {
```

Comment:
I don't think 50 is large enough to detect obstacles in time. Try running into an obstacle at 100% speed with your algorithm in SMCE.

Reply:
Which distance do you suggest?

Reply:
I suggest you try different values in SMCE yourself. I used 200 before, but that can be fine-tuned further.

Reply:
Optimally you would compute the distance on the fly depending on the current speed of the car, but that might be too much for now. :)